pandas/core/internals/blocks.py

```diff
@@ -1746,6 +1747,27 @@ def _slice(self, slicer):
 
         return self.values[slicer]
 
+    def _try_cast_result(self, result, dtype=None):
+        """
+        if we have an operation that operates on for example floats
+        we want to try to cast back to our EA here if possible
+
+        result could be a 2-D numpy array, e.g. the result of
+        a numeric operation; but it must be shape (1, X) because
+        we by-definition operate on the ExtensionBlocks one-by-one
+
+        result could also be an EA Array itself, in which case it
+        is already a 1-D array
+        """
+        try:
+
+            result = self._holder._from_sequence(
+                np.asarray(result).ravel(), dtype=dtype)
```

Comment:
I'm a bit concerned by the `asarray` here. Is that just so we can do the `.ravel`?

Consider a silly example like

```python
df.groupby('key').apply(lambda x: x.array)
```

Will that end up hitting this, and so calling `asarray` and converting to ndarray?

Reply:
does this make sense?
```
df.groupby('A').B.apply(lambda x: x.array)
```

```
(Pdb) p df
   A                         B
0  1 2000-01-01 18:00:00-06:00
1  1 2000-01-01 18:00:00-06:00
2  2                       NaT
3  2                       NaT
4  3 1999-12-31 18:00:00-06:00
5  3 1999-12-31 18:00:00-06:00
6  1 2000-01-01 18:00:00-06:00
7  4 2000-01-02 18:00:00-06:00
(Pdb) p result
A
1    [2000-01-01 18:00:00-06:00, 2000-01-01 18:00:0...
2                                           [NaT, NaT]
3    [1999-12-31 18:00:00-06:00, 1999-12-31 18:00:0...
4                          [2000-01-02 18:00:00-06:00]
Name: B, dtype: object
```